File: apps/institution-api/Models/Institution.cs
using System;
using System.Collections.Generic;

#nullable disable

namespace ProviderApi.Models
{
    public partial class Institution
    {
        public Institution()
        {
            InstitutionAddresses = new HashSet<InstitutionAddress>();
            InstitutionEmails = new HashSet<InstitutionEmail>();
            InstitutionLocations = new HashSet<InstitutionLocation>();
            InstitutionPhones = new HashSet<InstitutionPhone>();
            InstitutionPublications = new HashSet<InstitutionPublication>();
            InstitutionWebsites = new HashSet<InstitutionWebsite>();
        }

        public int InstitutionId { get; set; }
        public string OperatingName { get; set; }
        public string InstitutionName { get; set; }
        public string InstitutionLegalName { get; set; }
        public string InstitutionShortName { get; set; }
        public string InstitutionType { get; set; }
        public string OperatingType { get; set; }
        public string HomeJurisdiction { get; set; }
        public string InstituitionCategory { get; set; }
        public string InstitutionCode { get; set; }
        public int PaprsinstitutionNumber { get; set; }
        public int? ProvincialInstitutionNumber { get; set; }
        public string FederalInstitutionNumber { get; set; }
        public string InstitutionStatus { get; set; }
        public DateTime BusinessStartDate { get; set; }
        public DateTime BusinessEndDate { get; set; }
        public Guid? InstitutionGuid { get; set; }

        public virtual ICollection<InstitutionAddress> InstitutionAddresses { get; set; }
        public virtual ICollection<InstitutionEmail> InstitutionEmails { get; set; }
        public virtual ICollection<InstitutionLocation> InstitutionLocations { get; set; }
        public virtual ICollection<InstitutionPhone> InstitutionPhones { get; set; }
        public virtual ICollection<InstitutionPublication> InstitutionPublications { get; set; }
        public virtual ICollection<InstitutionWebsite> InstitutionWebsites { get; set; }
    }
}

File: libs/common/models/institution.ts


import { InstitutionAddress } from './institution-address';
import { InstitutionEmail } from './institution-email';
import { InstitutionLocation } from './institution-location';
import { InstitutionPhone } from './institution-phone';
import { InstitutionPublication } from './institution-publication';
import { InstitutionWebsite } from './institution-website';

export class Institution {
    public institutionId: number = 0;
    public operatingName: string = '';
    public institutionName: string = '';
    public institutionLegalName: string = '';
    public institutionShortName: string = '';
    public institutionType: string = '';
    public operatingType: string = '';
    public homeJurisdiction: string = '';
    public instituitionCategory: string = '';
    public institutionCode: string = '';
    public paprsinstitutionNumber: number = 0;
    public provincialInstitutionNumber: number;
    public federalInstitutionNumber: string = '';
    public institutionStatus: string = '';
    public businessStartDate: string = '';
    public businessEndDate: string = '';
    public institutionGuid: string;
    public institutionAddresses: InstitutionAddress[] = [];
    public institutionEmails: InstitutionEmail[] = [];
    public institutionLocations: InstitutionLocation[] = [];
    public institutionPhones: InstitutionPhone[] = [];
    public institutionPublications: InstitutionPublication[] = [];
    public institutionWebsites: InstitutionWebsite[] = [];

    constructor(init? : Partial<Institution>) {
        
        Object.assign(this, init);
    }
}

